drivers/video/mt9m114.c

```diff
@@ -461,12 +487,36 @@ static int mt9m114_get_caps(const struct device *dev, enum video_endpoint_id ep,
 	return 0;
 }
 
+static int mt9m114_set_ctrl(const struct device *dev, unsigned int cid, void *value)
+{
+	int ret = 0;
+
+	switch (cid) {
+	case VIDEO_CID_HFLIP:
+		ret = mt9m114_set_orientation(dev, (int)value, MT9M114_CAM_CONTROL_HORIZ_FLIP_EN);
```

Comment:
`mt9m114_modify_reg(dev, MT9M114_CAM_SENSOR_CTRL_READ_MODE, MT9M114_CAM_SENSOR_CTRL_HORZ_FLIP_EN, (int)value ? MT9M114_CAM_SENSOR_CTRL_HORZ_FLIP_EN, 0);`

Reply:
I see your point, but it is not feasible right now due to the definition of `mt9m114_modify_reg` function which works only with 8 bit registers. In this case, the `MT9M114_CAM_SENSOR_CTRL_READ_MODE` register is 16 bit.

I see two alternatives:
1. rewrite function to work with 8, 16 and 32 bit registers
2. write a function for each register size

Then, it would be feasible

Reply:
Ah, sorry, I just doublechecked. Indeed, inside `mt9m114_modify_reg` , it calls read/write reg on 8 bits only

	uint8_t newVal;
	int ret = mt9m114_read_reg(dev, addr, sizeof(oldVal), &oldVal);

Reply:
So if possible, could you modify also `mt9m114_modify_reg` so that it supports both 8 bits and 16 bits ? then `mt9m114_modify_reg` is only used here and on MT9M114_RST_AND_MISC_CONTROL in software reset so the changes will not be much ?

Reply:
But it's not mandatory ... If you feel uncomfortatble, it's ok for me to let it as is for now. I will make the change later

Reply:
No problem! Actually, I was thinking about it, I will do it :)

Reply:
@ngphibang I have pushed a new version with a common `mt9m114_modify_reg` function:

```c
static int mt9m114_modify_reg(const struct device *dev, const uint16_t addr, 
			      uint8_t reg_size, const uint32_t mask, const uint32_t val)
{
	uint32_t oldVal = 0;
	uint32_t newVal = 0;

	int ret = mt9m114_read_reg(dev, addr, reg_size, &oldVal);

	if (ret) {
		return ret;
	}

	newVal = (oldVal & ~mask) | (val & mask);

	return mt9m114_write_reg(dev, addr, reg_size, &newVal);
}
```

The idea is to cast the value to the larger likely value (4 bytes). Then, the value is correctly read and written since they are pass as void pointer and interpreated correctly in both the read and write functions. I have check it out and works correctly, what do you think?

Reply:
Maybe it becomes interesting to have a [CCI (Camera Common Interface) library](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/media/v4l2-core/v4l2-cci.c) that has read/write/modify built-in for various sizes.

Reply:
I think that it is beyond the scope of this pull request, but it is something interested. However, right now we are using only an i2c bus. Thus, it would be interesting in case that we can configure cameras with other buses like SPI, otherwise, I don't see the point.

Reply:
Yes absolutely, not related to this PR! I am thinking in the background: someone with a camera sensor knows the I2C address to toggle a feature and wants it upstream: how to make it just a 1-line diff to facilitate contribution.

I agree though: good not to introduce unneeded abstraction.

Reply:
@josuah I think adding CCI would be helpful. But even with CCI, in Linux, camera drivers still uses different i2c APIs to read / write registers, so it will not make it a "standard way" for every drivers